File: src/app/common/purchase.ts
import { Address } from "./address";
import { Customer } from "./customer";
import { Order } from "./order";
import { OrderItem } from "./order-item";

export class Purchase {
    customer: Customer
    order: Order;
    shippingAddress: Address;
    billingAddress: Address;
    orderItems: OrderItem[];

}

File: 02-backend/spring-boot-ecommerce/src/main/java/com/rochak/springbootecommerce/dto/Purchase.java
package com.rochak.springbootecommerce.dto;

import com.rochak.springbootecommerce.entity.Address;
import com.rochak.springbootecommerce.entity.Customer;
import com.rochak.springbootecommerce.entity.Order;
import com.rochak.springbootecommerce.entity.OrderItem;
import lombok.Data;

import java.util.Set;

@Data
public class Purchase {

    private Customer customer;

    private Address shippingAddress;

    private Address billingAddress;

    private Order order;

    private Set<OrderItem> orderItems;
}
